ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -523,3 +548,23 @@ decodeShelleyResult query = case query of
     GetStakeDistribution         -> fromCBOR
     GetFilteredUTxO {}           -> fromCBOR
     GetUTxO                      -> fromCBOR
+    GetCurrentLedgerState        -> fromCborInCbor
+
+-- We encode some items like LedgerState as CBOR-in-CBOR to avoid having to version
+-- them.
+-- Code receiving a CBOR-in-CBOR encode value can always safely decode the outer
+-- ByteString. It can then try to CBOR decode the inner ByteString and if that fails
+-- can alway do a generic CBOR decode for debug purposes.
+
+fromCborInCbor :: FromCBOR a => Decoder s a
+fromCborInCbor = do
+  tag <- CBOR.decodeWord8
+  when (tag /= 24) $ fail "expected tag 24 (CBOR-in-CBOR)"
+  fromCBOR
+
+toCborInCbor :: ToCBOR a => a -> Encoding
+toCborInCbor x =
+  mconcat
+    [ toCBOR (24 :: Word8)
+    , toCBOR x
+    ]
```

Comment:
As commented elsewhere: I prefer we use `wrapCBORinCBOR toCBOR` instead of copy-pasting its implementation here.

Reply:
The types of `wrapCBORinCBOR` and `toCborInCbor` are different as are their implementations!

Reply:
```
wrapCBORinCBOR :: (a -> Encoding) -> a -> Encoding
```
and
```
toCborInCbor :: ToCBOR a => a -> Encoding
```